### PR TITLE
Enhance the wavefront-sdk-go to support Kubernetes environments

### PR DESCRIPTION
**Pull request by laullon**

### CODE REVIEW

File: application/application.go [hunk 1]
```diff
@@ -23,15 +30,44 @@ func New(application, service string) Tags {
 }
 
 // Map with all values
-func (a Tags) Map() map[string]string {
+func (app *Tags) Map() map[string]string {
+	os.Getenv("pp")
```
Comment (vikramraman) [inline application/application.go:34]: is this necessary? remove?

Comment (laullon) [reply application/application.go:34]: oh, sorry...  ;-)

File: application/application.go [hunk 2]
```diff
@@ -23,15 +30,44 @@ func New(application, service string) Tags {
...
 		allTags[k] = v
 	}
 	return allTags
 }
+
+// AddCustomTagsFromEnv set additional custom tags from environment variables that match the given regex.
+func (app *Tags) AddCustomTagsFromEnv(regx string) {
+	r, err := regexp.Compile(regx)
+	if err != nil {
+		log.Printf("Error creating custom tags: %v\n", err)
```
Comment (vikramraman) [inline application/application.go:50]: Return an error here rather than just log it?

File: application/application.go [hunk 3]
```diff
@@ -23,15 +30,43 @@ func New(application, service string) Tags {
...
+			if len(v) > 0 {
+				app.CustomTags[k] = v
+			}
+		}
+	}
+}
+
+// AddCustomTagFromEnv Set a custom tag from the given environment variable.
+func (app *Tags) AddCustomTagFromEnv(varName, tag string) {
+	v := os.Getenv(varName)
```
Comment (vikramraman) [inline application/application.go:68]: Maybe return an error if the env var doesn't exist?

**Pull request merged**